control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -185,10 +185,62 @@ impl ResourceResize for OperationGuardArc<VolumeSpec> {
 
     async fn resize(
         &mut self,
-        _registry: &Registry,
-        _request: &Self::Resize,
+        registry: &Registry,
+        request: &Self::Resize,
     ) -> Result<Self::ResizeOutput, SvcError> {
-        unimplemented!()
+        let specs = registry.specs();
+        let spec = self.as_ref().clone();
+        let state = registry.volume_state(&request.uuid).await?;
+
+        // Pre-Check - volume state ok for resize.
+        if !spec.vol_status_ok_for_resize() {
+            return Err(SvcError::InUse {
+                kind: stor_port::transport_api::ResourceKind::Volume,
+                id: spec.uuid.to_string(),
+            });
+        }
+
+        let spec_clone = self
+            .start_update(
+                registry,
+                &state,
+                VolumeOperation::Resize(request.requested_size),
+            )
+            .await?;
+        // For the volume being resized, resize each replica of the volume. For a failure to
+        // any of the replica, the volume resize operation is to be deemed failure.
+        let replicas = specs.volume_replicas(&request.uuid);
+        for replica in replicas {
+            let mut replica = replica.operation_guard_wait().await?;
```

Comment:
you need to move this to a sub function, cannot return error while update has been started..

Reply:
Do you mean something like this? `resize_volume_replicas` is new helper function.

```
        let spec_clone = volume.start_update().await?;
        match self.resize_volume_replicas(registry, &replicas, request.requested_size, &spec_clone).await {
            Ok(_) => {
                let volume = registry.volume(&request.uuid).await?;
                self.complete_update(registry, Ok(volume), spec_clone).await
            },
            Err(e) => Err(e),
        }
```

Reply:
In line 214 I don't think you can return using the ?, there are 2 things we can do, i.e if 214 gives an error `continue`, `ignore error silently/log and move ahead` or use a `validate_update_step` to return the error from 214.

Reply:
Alright, I'm moving these lines to a helper function as mentioned above, will handle there.

Reply:
@Abhinandan-Purkait @tiagolobocastro PTAL if the diff looks ok now.